src/sentry/api/endpoints/group_event_details.py

```diff
@@ -37,11 +38,20 @@ def get(self, request: Request, group: Group) -> Response:
         :pparam string group_id: the ID of the issue
         """
         environments = [e.name for e in get_environments(request, group.project.organization)]
-
-        event = group.get_latest_event_for_environments(environments)
-
-        if not event:
-            return Response({"detail": "No events found for group"}, status=404)
+        event = None
+
+        if event_id == "latest":
+            event = group.get_latest_event_for_environments(environments)
+        elif event_id == "oldest":
+            event = group.get_oldest_event_for_environments(environments)
+        else:
+            event = eventstore.get_event_by_id(group.project.id, event_id, group_id=group.id)
+            # TODO: Remove `for_group` check once performance issues are moved to the issue platform
+            if hasattr(event, "for_group") and event.group:
```

Comment:
This logic is taken from the project event details endpoint: https://github.com/getsentry/sentry/blob/6550621af105961899e897dfe278e0468d5764be/src/sentry/api/endpoints/project_event_details.py#L88-L90